edx_solutions_api_integration/courses/views.py

```diff
@@ -2659,11 +2659,14 @@ def post(self, request):
         response_data = []
         for course_structure in course_structures:
             blocks = course_structure.structure.get('blocks', {})
+
+            course = None
             for block in blocks.values():
                 block['name'] = block.pop('display_name')
                 block['category'] = block.pop('block_type')
+                if block['category'] == 'course' and course is None:
```

Comment:
@nasirhjafri  one small change required here, can we check the course first like:
```
if course is None and block['category'] == 'course':
```